### PR TITLE
feat(components/forms): deprecate `SkyFileAttachmentsModule`; create `SkyFileAttachmentModule` and `SkyFileDropModule`

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
[AB#2916702](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2916702)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2585/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2585/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2585/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2585/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2585/playground/)

File: libs/components/forms/src/index.ts [hunk 1]
```diff
@@ -5,30 +5,32 @@ export { SkyCheckboxGroupHeadingLevel } from './lib/modules/checkbox/checkbox-gr
 export { SkyCheckboxGroupHeadingStyle } from './lib/modules/checkbox/checkbox-group-heading-style';
 export { SkyCheckboxModule } from './lib/modules/checkbox/checkbox.module';
 
-export { SkyFieldGroupModule } from './lib/modules/field-group/field-group.module';
 export { SkyFieldGroupHeadingLevel } from './lib/modules/field-group/field-group-heading-level';
 export { SkyFieldGroupHeadingStyle } from './lib/modules/field-group/field-group-heading-style';
+export { SkyFieldGroupModule } from './lib/modules/field-group/field-group.module';
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/index.ts:10]: 👇 Alphabetical ordering.

File: apps/code-examples/src/app/code-examples/forms/file-attachment/basic/demo.component.html [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
-  [required]="required"
-  [stacked]="stacked"
-  [validateFn]="validateFile"
-  (filesChanged)="onFilesChanged($event)"
-  (linkChanged)="onLinkChanged($event)"
-/>
-<div *ngFor="let file of allItems">
-  <sky-file-item [fileItem]="file" (deleteFile)="deleteFile($event)" />
-</div>
+<form [formGroup]="formGroup" class="sky-padding-even-lg">
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/forms/file-attachment/basic/demo.component.html:1]: Code example contents were not changed; the file names were changed to match the new module names: `file-attachment` and `file-drop`.

File: libs/components/forms/src/lib/modules/file-attachment/shared/file-size.pipe.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { DecimalPipe } from '@angular/common';
+import { formatNumber } from '@angular/common';
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/shared/file-size.pipe.ts:1]: We were injecting `DecimalPipe` which was causing a few problems with undefined injectors with the new standalone components. We're only using `DecimalPipe` to format the number; Angular exports the same utility function `formatNumber` in its public API, so we can forego injecting `DecimalPipe` altogether.

File: libs/components/forms/src/index.ts [hunk 4]
```diff
@@ -5,30 +5,32 @@ export { SkyCheckboxGroupHeadingLevel } from './lib/modules/checkbox/checkbox-gr
...
 
-export { SkyFormErrorsModule } from './lib/modules/form-error/form-errors.module';
 export { SkyFormErrorModule } from './lib/modules/form-error/form-error.module';
+export { SkyFormErrorsModule } from './lib/modules/form-error/form-errors.module';
 
 export { SkyInputBoxHostService } from './lib/modules/input-box/input-box-host.service';
 export { SkyInputBoxPopulateArgs } from './lib/modules/input-box/input-box-populate-args';
 export { SkyInputBoxModule } from './lib/modules/input-box/input-box.module';
 
+export { SkyRadioModule } from './lib/modules/radio/radio.module';
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/forms/src/index.ts:30]: This was already in alphabetical order

File: libs/components/forms/src/index.ts [hunk 5]
```diff
@@ -48,21 +50,21 @@ export { SkyFormFieldLabelTextRequiredService } from './lib/modules/shared/form-
 // Obscure names are used to indicate types are not part of the public API.
 
 export { SkyCharacterCounterIndicatorComponent as λ1 } from './lib/modules/character-counter/character-counter-indicator.component';
-export { SkyCharacterCounterInputDirective as λ2 } from './lib/modules/character-counter/character-counter.directive';
 export { SkyCharacterCounterScreenReaderPipe as λ23 } from './lib/modules/character-counter/character-counter-screen-reader.pipe';
+export { SkyCharacterCounterInputDirective as λ2 } from './lib/modules/character-counter/character-counter.directive';
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/forms/src/index.ts:54]: This was also

File: libs/components/forms/src/index.ts [hunk 6]
```diff
@@ -48,21 +50,21 @@ export { SkyFormFieldLabelTextRequiredService } from './lib/modules/shared/form-
...
+export { SkyCheckboxGroupComponent as λ24 } from './lib/modules/checkbox/checkbox-group.component';
 export { SkyCheckboxLabelComponent as λ4 } from './lib/modules/checkbox/checkbox-label.component';
 export { SkyCheckboxComponent as λ3 } from './lib/modules/checkbox/checkbox.component';
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/forms/src/index.ts:57]: `SkyCheckboxComponent` should be at line 55

File: libs/components/forms/src/index.ts [hunk 7]
```diff
@@ -48,21 +50,21 @@ export { SkyFormFieldLabelTextRequiredService } from './lib/modules/shared/form-
...
+export { SkyFileAttachmentLabelComponent as λ6 } from './lib/modules/file-attachment/file-attachment/file-attachment-label.component';
+export { SkyFileAttachmentComponent as λ7 } from './lib/modules/file-attachment/file-attachment/file-attachment.component';
+export { SkyFileDropComponent as λ8 } from './lib/modules/file-attachment/file-drop/file-drop.component';
+export { SkyFileItemComponent as λ9 } from './lib/modules/file-attachment/file-drop/file-item.component';
 export { SkyFormErrorComponent as λ22 } from './lib/modules/form-error/form-error.component';
+export { SkyFormErrorsComponent as λ21 } from './lib/modules/form-error/form-errors.component';
 export { SkyInputBoxControlDirective as λ20 } from './lib/modules/input-box/input-box-control.directive';
-export { SkyInputBoxComponent as λ10 } from './lib/modules/input-box/input-box.component';
 export { SkyInputBoxHintTextPipe as λ26 } from './lib/modules/input-box/input-box-hint-text.pipe';
+export { SkyInputBoxComponent as λ10 } from './lib/modules/input-box/input-box.component';
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/forms/src/index.ts:67]: This too

File: libs/components/forms/src/lib/modules/file-attachment/shared/file-item.service.ts [hunk 8]
```diff
@@ -5,7 +5,7 @@ import { SkyFileItem } from './file-item';
 /**
  * @internal
  */
-@Injectable()
+@Injectable({ providedIn: 'root' })
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/forms/src/lib/modules/file-attachment/shared/file-item.service.ts:8]: Do we need to call this out in the changes?

File: libs/components/forms/src/index.ts (hunk 4 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/forms/src/index.ts:30]: The ordering is done first against the file's path, and then the types that are exported by the file (which is desired). This was the result of executing the VS Code "Organize Imports" command.

File: libs/components/forms/src/lib/modules/file-attachment/shared/file-item.service.ts (hunk 8 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/forms/src/lib/modules/file-attachment/shared/file-item.service.ts:8]: No, since it's an internal-only service.

**Review by Blackbaud-CoreyArcher — APPROVED** [review]

**Pull request merged**